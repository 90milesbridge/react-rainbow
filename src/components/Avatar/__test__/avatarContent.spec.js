import React from 'react';
import { mount } from 'enzyme/build/index';
import renderer from 'react-test-renderer';
import AvatarContent from './../avatarContent';

describe('<AvatarContent/>', () => {
    it('should render an img when src is passed', () => {
        const component = mount(
            <AvatarContent iconName="standard:user" src="images/avatar1.jpg" />,
        );
        expect(component.find('img'));
    });
    it('should pass right props when a valid src is passed', () => {
        const component = mount(
            <AvatarContent iconName="standard:user" src="images/avatar1.jpg" title="Title" />,
        );
        expect(component.find('img').props()).toEqual({
            src: 'images/avatar1.jpg',
            onError: expect.any(Function),
            title: 'Title',
        });
    });
    it('should render an abbr when src is not passed and initials props is passed', () => {
        const component = mount(
            <AvatarContent iconName="standard:user" initials="JD" />,
        );
        expect(component.find('abbr'));
    });
    it('should pass right props when src is not passed and initials props is passed', () => {
        const component = mount(
            <AvatarContent iconName="standard:user" initials="JD" title="Title" />,
        );
        expect(component.find('abbr').props()).toEqual({
            children: 'JD',
            className: 'slds-avatar__initials slds-icon-standard-user',
            title: 'Title',
        });
    });
    it('should render an Icon when src and initials are not passed', () => {
        const component = mount(
            <AvatarContent iconName="standard:user" />,
        );
        expect(component.find('Icon'));
    });
    it('should render an Icon when src is not passed and invalid initials is passed', () => {
        const component = mount(
            <AvatarContent iconName="standard:user" initials="jd" />,
        );
        expect(component.find('Icon'));
    });
    it('should pass right props to Icon component', () => {
        const component = mount(
            <AvatarContent iconName="standard:user" title="Title" />,
        );
        expect(component.find('Icon').prop('iconName')).toBe('standard:user');
        expect(component.find('Icon').prop('title')).toBe('Title');
    });
    it('should have the right class names', () => {
        const component = renderer.create(
            <AvatarContent iconName="standard:user" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when initials variant is default', () => {
        const component = renderer.create(
            <AvatarContent iconName="standard:user" initials="JD" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when initials variant is inverse', () => {
        const component = renderer.create(
            <AvatarContent iconName="standard:user" initials="JD" initialsVariant="inverse" />,
        );
        expect(component).toMatchSnapshot();
    });
});
