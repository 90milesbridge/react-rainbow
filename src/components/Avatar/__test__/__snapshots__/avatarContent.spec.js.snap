// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarContent/> should have the right class names 1`] = `
<span
  className="slds-icon_container slds-icon-standard-user"
  style={Object {}}
>
  <svg
    aria-hidden={true}
    className="slds-icon"
    style={Object {}}
  >
    <use
      xlinkHref="/icons/standard-sprite/svg/symbols.svg#user"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    />
  </svg>
</span>
`;

exports[`<AvatarContent/> should have the right class names when initials variant is default 1`] = `
<abbr
  className="slds-avatar__initials slds-icon-standard-user"
>
  JD
</abbr>
`;

exports[`<AvatarContent/> should have the right class names when initials variant is inverse 1`] = `
<abbr
  className="slds-avatar__initials slds-avatar__initials_inverse"
>
  JD
</abbr>
`;
